pkg/virt-controller/watch/migration_test.go

```diff
@@ -2162,6 +2161,59 @@ var _ = Describe("Migration watcher", func() {
 			expectTargetPodWithSELinuxLevel(vmi.Namespace, vmi.UID, migration.UID, "")
 		})
 	})
+
+	Context("Secondary queue", func() {
+		It("should properly flush when a migration finishes", func() {
+			By("Creating 1 pending migration. It will be picked up by the first call to Execute()")
+			vmi := newVirtualMachine("testvmipending", virtv1.Running)
+			migration := newMigration("testmigrationpending", vmi.Name, virtv1.MigrationPending)
+			addMigration(migration)
+			addVirtualMachineInstance(vmi)
+			addPod(newSourcePodForVirtualMachine(vmi))
+
+			By("Creating a migration that's ready to complete. It will be picked up by the second call to Execute()")
+			vmi = newVirtualMachine("testvmi", virtv1.Running)
+			addNodeNameToVMI(vmi, "node02")
+			migration = newMigration("testmigration", vmi.Name, virtv1.MigrationRunning)
+			targetPod := newTargetPodForVirtualMachine(vmi, migration, k8sv1.PodPending)
+			targetPod.Spec.NodeName = "node01"
+			vmi.Status.MigrationState = &virtv1.VirtualMachineInstanceMigrationState{
+				MigrationUID:                   migration.UID,
+				TargetNode:                     "node01",
+				SourceNode:                     "node02",
+				TargetNodeAddress:              "10.10.10.10:1234",
+				StartTimestamp:                 pointer.P(metav1.Now()),
+				EndTimestamp:                   pointer.P(metav1.Now()),
+				TargetNodeDomainReadyTimestamp: pointer.P(metav1.Now()),
+				Failed:                         false,
+				Completed:                      true,
+			}
+			addMigration(migration)
+			addVirtualMachineInstance(vmi)
+			addPod(newSourcePodForVirtualMachine(vmi))
+			addPod(targetPod)
+
+			By("Creating 5 running migrations")
+			for i := 0; i < 5; i++ {
+				vmi := newVirtualMachine(fmt.Sprintf("testvmi%d", i), virtv1.Running)
+				migration := newMigration(fmt.Sprintf("testmigration%d", i), vmi.Name, virtv1.MigrationRunning)
+				pod := newTargetPodForVirtualMachine(vmi, migration, k8sv1.PodRunning)
+				addMigration(migration)
+				addVirtualMachineInstance(vmi)
+				addPod(pod)
+			}
+
+			By("Executing the controller and expecting the pending migration in the secondary queue")
+			controller.Execute()
+			Expect(controller.pendingQueue).To(Equal([]string{"default/testmigrationpending"}))
```

Comment:
don't we need to lock before reading?

Reply:
I don't think so. The only thing that touches `pendingQueue` is `Execute()`, which we run manually in our own thread.
There's no controller loop running and each test has its own controller.
The only concern could be about the informer, but it doesn't touch `pendingQueue` (and has no reason to trigger right now anyway).
Or am I missing something?